File: via_l2g/递增的三元子序列.cs
public class Solution {
    public bool IncreasingTriplet(int[] nums) {
        int MAX = 10000000;
        int first = MAX;
        int second = MAX;
        int ext = MAX;
        foreach(int x in nums) {
            if (ext != MAX) {
                if (x > ext && x < second) {
                    first = ext;
                    second = x;
                    ext = MAX;
                    continue;
                }
            }
            if (first == MAX) first = x;
            if (second != MAX) {
                if (x > second) return true;
                if (x < second && x > first) second = x;
                if (x < first) ext = x;
            }
            else {
                if (x <= first) first = x;
                else second = x;
            }
        }
        return false;
    }
}

File: via_l2g/递增的三元子序列.py
class Solution:
    def increasingTriplet(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        first = None
        second = None
        ext = None
        for x in nums:
            if ext is not None:
                if ext < x < second:
                    first, second = ext, x
                    ext = None
                    continue
            if first is None:
                first = x
            if second is not None:
                if x > second:
                    return True
                if first < x < second:
                    second = x
                if x < first:
                    ext = x
            else:
                if x <= first:
                    first = x
                else:
                    second = x
        return False
